solidity/contracts/bridge/MovingFunds.sol

```diff
@@ -521,4 +531,376 @@ library MovingFunds {
+            bytes32 firstInputOutpointTxHash,
+            uint32 firstInputOutpointIndex,
+            uint256 firstInputLength
+        ) = parseMovedFundsSweepTxInputAt(
+                sweepTxInputVector,
+                inputStartingIndex
+            );
+
+        // Build the request key and fetch the corresponding moved funds sweep
+        // request from contract storage.
+        MovedFundsSweepRequest storage sweepRequest = self
+            .movedFundsSweepRequests[
+                uint256(
+                    keccak256(
+                        abi.encodePacked(
+                            firstInputOutpointTxHash,
+                            firstInputOutpointIndex
+                        )
+                    )
+                )
+            ];
+
+        // The sweep request must exist, must be not processed yet, and must
+        // belong to the sweeping wallet.
+        require(sweepRequest.createdAt != 0, "Sweep request does not exist");
+        require(sweepRequest.sweptAt == 0, "Sweep request already processed");
+        require(
+            sweepRequest.walletPubKeyHash == walletPubKeyHash,
+            "Sweep request belongs to another wallet"
+        );
+        // If the validation passed, the sweep request must be marked as
+        // processed and its value should be counted into the total inputs
+        // value sum.
+        /* solhint-disable-next-line not-rely-on-time */
+        sweepRequest.sweptAt = uint32(block.timestamp);
+        inputsTotalValue += sweepRequest.value;
+
+        // TODO: Decrease the sweep request count for the sweeping wallet.
+        //       That will be handled in the PR that will block moving
+        //       funds commitments for wallets with pending sweep requests.
+
+        // If the main UTXO for the sweeping wallet exists, it must be processed.
+        if (mainUtxo.txHash != bytes32(0)) {
+            // The second input is supposed to point to that sweeping wallet
+            // main UTXO. We need to parse that input.
+            (
+                bytes32 secondInputOutpointTxHash,
+                uint32 secondInputOutpointIndex,
+
+            ) = parseMovedFundsSweepTxInputAt(
+                    sweepTxInputVector,
+                    inputStartingIndex + firstInputLength
+                );
+            // Make sure the second input refers to the sweeping wallet main UTXO.
+            require(
+                mainUtxo.txHash == secondInputOutpointTxHash &&
+                    mainUtxo.txOutputIndex == secondInputOutpointIndex,
+                "Second input must point to the wallet's main UTXO"
+            );
+
+            // If the validation passed, count the main UTXO value into the
+            // total inputs value sum.
+            inputsTotalValue += mainUtxo.txOutputValue;
+
+            // Main UTXO used as an input, mark it as spent. This is needed
+            // to defend against fraud challenges referring to this main UTXO.
```

Comment:
It's not even that there is no need to defend. That UTXO was honestly spent. One may still challenge the wallet but the wallet can defeat the challenge by having that UTXO marked as honestly spent.

Reply:
Yeah, but from the code perspective, without that explanation, someone can wonder why do we need to add this UTXO to the `spentMainUtxos` mapping at all.